mentor_classifier_api/src/mentor_classifier_api/__init__.py

```diff
@@ -4,39 +4,158 @@
 #
+import os
+import logging
+import json
 from dotenv import load_dotenv
 
 load_dotenv()  # take environment variables from .env.
 
 from logging.config import dictConfig  # NOQA
 
-from flask import Flask  # NOQA
+from flask import Flask, has_request_context, g, request  # NOQA
 from flask_cors import CORS  # NOQA
 from .config_default import Config  # NOQA
 
+if os.environ.get("IS_SENTRY_ENABLED", "") == "true":
+    import sentry_sdk  # NOQA E402
+    from sentry_sdk.integrations.flask import FlaskIntegration  # NOQA E402
+
+
+class JSONFormatter(logging.Formatter):
+    RECORD_ATTRS = [
+        "name",
+        "levelname",
+        "filename",
+        "module",
+        "path",
+        "lineno",
+        "funcName",
+        "endpoint",
+        "method",
+        "url",
+        "headers",
+    ]
+
+    def to_payload(self, record):
+        payload = {
+            attr: getattr(record, attr)
+            for attr in self.RECORD_ATTRS
+            if hasattr(record, attr)
+        }
+        # make sure log messages are consistent across services:
+        payload["level"] = payload["levelname"].lower()
+        del payload["levelname"]
+        payload["logger"] = payload["name"]
+        del payload["name"]
+        payload["message"] = record.getMessage()
+        return payload
+
+    def to_json(self, payload):
+        # do not assume there's a Flask request context here so must use FLASK_ENV env var not app.debug
+        indent = 2 if os.environ.get("FLASK_ENV", "") == "development" else None
+        return json.dumps(payload, indent=indent)
+
+    def format(self, record):
+        payload = self.to_payload(record)
+        return self.to_json(payload)
+
+
+class RequestJSONFormatter(JSONFormatter):
+    def format(self, record):
+        if has_request_context():
+            # todo for distributed tracing:
+            # record.request_id = g.request_id if hasattr(g, 'request_id') else '-'
+            if hasattr(g, "response_time"):
+                setattr(record, "response-time", g.response_time)
+            record.path = request.full_path
+            record.endpoint = request.endpoint
+            record.method = request.method
+            record.url = request.url
+            # make sure to redact sensitive info: cookies, auth...
+            record.headers = {
+                k: v for k, v in request.headers.items() if "auth" not in k.lower()
+            }
+
+        return super().format(record)
+
+
+class RequestFilter(logging.Filter):
+
+    # def __init__(self, methods=None):
+    #     self.methods = methods or []
+    #     super().__init__()
+
+    def filter(self, record):
+        # TODO redact sensitive data
+        return True
+        # if hasattr(record, 'method'):
+        #     if record.method in self.methods:
+        #         return True
+        # else:
+        #     return True
+
```

Comment:
you can remove this one, my idea was to redact some sensitive data (tokens, passwords, ...), but didn't have time to implement it